tests/flow/test_list.py

```diff
@@ -1421,5 +1421,427 @@ def test13_dedup(self):
 
         expected_result = [[3,[1,2],[1]]]
         query = """RETURN list.dedup([3,[1,2],3,[1],[1,2]])"""
+
+    def test14_union(self):
+        # 2nd arg should be list
+        try:
+            redis_graph.query("RETURN list.union([1,2,3], '2', 1)")
+            self.env.assertTrue(False)
+        except ResponseError as e:
+            self.env.assertContains("Type mismatch: expected List or Null but was String", str(e))
+
+        # 3rd arg should be Integer
+        try:
+            redis_graph.query("RETURN list.union([1,2,3], [2], '1')")
+            self.env.assertTrue(False)
+        except ResponseError as e:
+            self.env.assertContains("Type mismatch: expected Integer but was String", str(e))
+
+        # Test without input argument
+        try:
+            query = """RETURN list.union()"""
+            redis_graph.query(query)
+            self.env.assertTrue(False)
+        except ResponseError as e:
+            self.env.assertContains("Received 0 arguments to function 'list.union', expected at least 2", str(e))
+
+        # Test with 1 input argument
+        try:
+            query = """RETURN list.union([1,2,3])"""
+            redis_graph.query(query)
+            self.env.assertTrue(False)
+        except ResponseError as e:
+            self.env.assertContains("Received 1 arguments to function 'list.union', expected at least 2", str(e))
+
+        # Test with 4 input argument
+        try:
+            query = """RETURN list.union([1,2,3], [2], 1, 1)"""
+            redis_graph.query(query)
+            self.env.assertTrue(False)
+        except ResponseError as e:
+            self.env.assertContains("Received 4 arguments to function 'list.union', expected at most 3", str(e))
+
+
+        ### Test valid inputs ###
+        # NULL, NULL input should return NULL
+        expected_result = [None]
+        query = """RETURN list.union(null, null, 2)"""
+        actual_result = redis_graph.query(query)
+        self.env.assertEquals(actual_result.result_set[0], expected_result)
+
+        # NULL, NULL input should return NULL
+        expected_result = [[1,2,3]]
+        query = """RETURN list.union([1,3,2], null)"""
+        actual_result = redis_graph.query(query)
+        self.env.assertEquals(actual_result.result_set[0], expected_result)
+
+        # NULL, NULL input should return NULL
+        expected_result = [[1,2,3]]
+        query = """RETURN list.union(null, [1,3,2], 0)"""
+        actual_result = redis_graph.query(query)
+        self.env.assertEquals(actual_result.result_set[0], expected_result)
+
+        expected_result = [[1,2,3,4,None]]
+        query = """RETURN list.union([null,1,3,2], [null,4,null])"""
+        actual_result = redis_graph.query(query)
+        self.env.assertEquals(actual_result.result_set[0], expected_result)
+
+        expected_result = [[1,2,3,4,7,None]]
+        query = """RETURN list.union([null,1,3,2], [null,4,null,7,2], 0)"""
```

Comment:
Foreach policy please declare a variable, e.g. `beg_policy = 0` and pass the variable for readability.
Expecting a variable for policies 0,1 and 2